ql/src/test/org/apache/hadoop/hive/ql/exec/repl/TestAtlasDumpTask.java

```diff
@@ -92,48 +90,40 @@ public class TestAtlasDumpTask {
+    when(work.getMetricCollector()).thenReturn(metricCollector);
+    when(conf.get(HiveConf.ConfVars.REPL_ATLAS_ENDPOINT.varname)).thenReturn("http://localhost:21000/atlas");
+    when(conf.get(HiveConf.ConfVars.REPL_ATLAS_REPLICATED_TO_DB.varname)).thenReturn("tgtDb");
+    when(conf.get(HiveConf.ConfVars.REPL_SOURCE_CLUSTER_NAME.varname)).thenReturn("srcCluster");
+    when(conf.get(HiveConf.ConfVars.REPL_TARGET_CLUSTER_NAME.varname)).thenReturn("tgtCluster");
+    when(conf.get(ReplUtils.DEFAULT_FS_CONFIG)).thenReturn("hdfs:tgtFsUri:8020");
+    when(work.getStagingDir()).thenReturn(new Path("hdfs://tmp:8020/staging"));
+    when(work.getSrcDB()).thenReturn("srcDB");
+    when(work.isBootstrap()).thenReturn(true);
+    ReplLoggerFactory replLoggerFactoryMock = mock(ReplLoggerFactory.class);
+    when(replLoggerFactoryMock.createAtlasDumpLogger(anyString(), anyString())).thenReturn(logger);
+    atlasDumpTask = new AtlasDumpTask(atlasRestClient, conf, work, replLoggerFactoryMock);
     AtlasDumpTask atlasDumpTaskSpy = Mockito.spy(atlasDumpTask);
-    Mockito.when(conf.getBoolVar(HiveConf.ConfVars.HIVE_IN_TEST_REPL)).thenReturn(true);
-    Logger logger = Mockito.mock(Logger.class);
-    Whitebox.setInternalState(ReplState.class, logger);
+    when(conf.getBoolVar(HiveConf.ConfVars.HIVE_IN_TEST_REPL)).thenReturn(true);
     Mockito.doReturn(0L).when(atlasDumpTaskSpy)
-      .dumpAtlasMetaData(Mockito.any(AtlasRequestBuilder.class), Mockito.any(AtlasReplInfo.class));
-    Mockito.doNothing().when(atlasDumpTaskSpy).createDumpMetadata(Mockito.any(AtlasReplInfo.class),
-                                                                  Mockito.any(Long.class));
+            .dumpAtlasMetaData(any(AtlasRequestBuilder.class), any(AtlasReplInfo.class));
+    Mockito.doNothing().when(atlasDumpTaskSpy).createDumpMetadata(any(AtlasReplInfo.class),
+            any(Long.class));
     int status = atlasDumpTaskSpy.execute();
     Assert.assertEquals(0, status);
-    ArgumentCaptor<String> replStateCaptor = ArgumentCaptor.forClass(String.class);
-    ArgumentCaptor<Object> eventCaptor = ArgumentCaptor.forClass(Object.class);
-    ArgumentCaptor<Object> eventDetailsCaptor = ArgumentCaptor.forClass(Object.class);
-    Mockito.verify(logger,
-        Mockito.times(2)).info(replStateCaptor.capture(),
-        eventCaptor.capture(), eventDetailsCaptor.capture());
-    Assert.assertEquals("REPL::{}: {}", replStateCaptor.getAllValues().get(0));
-    Assert.assertEquals("ATLAS_DUMP_START", eventCaptor.getAllValues().get(0));
-    Assert.assertEquals("ATLAS_DUMP_END", eventCaptor.getAllValues().get(1));
-    Assert.assertTrue(eventDetailsCaptor.getAllValues().get(1).toString(), eventDetailsCaptor.getAllValues().get(0)
-        .toString().contains("{\"dbName\":\"srcDB\",\"dumpStartTime"));
-    Assert.assertTrue(eventDetailsCaptor
-        .getAllValues().get(1).toString().contains("{\"dbName\":\"srcDB\",\"dumpEndTime\""));
```

Comment:
Where are we validating these things now?

Reply:
Tests are refactored based on TestReplicationScenarios, by using log appenders.